extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -769,6 +760,32 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
   }
 
   /**
+   * Initializes a RefreshManager under the following conditions:
+   *   1. One has not already been initialized.
+   *   2. SRA is not enabled.
+   *   3. The slot is not inside an ad container (with the exception of
+   *      carousel and sticky ad).
+   *   4. The publisher has provided an appropriate refresh interval (>= 30s).
+   */
+  initRefreshManagerIfEligible_() {
+    let refreshInterval = 0;
+    this.refreshManager_ = this.refreshManager_ ||
+        // Not compatible with SRA
+        (!this.useSra &&
+         // Not compatible with container types except carousel and sticky-ad
+         !getEnclosingContainerTypes(this.element).filter(container =>
+           container != ValidAdContainerTypes['AMP-CAROUSEL'] &&
+           container != ValidAdContainerTypes['AMP-STICKY-AD']).length &&
+         // Publisher must supply a refresh interval >= 30s
+         (refreshInterval = getPublisherSpecifiedRefreshInterval(
```

Comment:
you could make a refreshmanager factory within refresh-manager code in which you place the interval check code where it would return null if invalid.  In fact it could take a verifying predicate and user().warn if invalid as I do not believe you're doing so right now if there is an invalid container type